### ISSUE TITLE
Noka N9 version of Maemo Leste

### ISSUE THREAD

**Issue by tvandorffy** [maemo-leste/bugtracker#535]
Dear developers,

The Nokia N9 image is more than two years old and it is unusable at the moment. (maemo-leste-1.0-armhf-n9-20190315.tar.gz)
Although it boots up, it  unresponsive and only the application button is showing and the rest is just a black screen:
https://www.youtube.com/watch?v=M3QsREPH7Is

If this is not due to installation problems (at the moment I do not think so),  would it be possible to refresh the N9 version and bring it up to the functionality of the VM versions?

Thanks,

tvandor

**Comment by MerlijnWajer** [comment]
When you say "up to the functionality of the VM versions", I presume you mean hildon-desktop booting ok, wifi and such working ok, but potentially not yet with PowerVR (3D) working?

**Comment by MerlijnWajer** [comment]
Ok, I'll check with @parazyd and see if we can get it going.

**Comment by tvandorffy** [comment]
> https://leste.maemo.org/Nokia_N9

Hi parazyd, I am sory, I do not follow... :-(

**Comment by parazyd** [comment]
Just left it here for reference.

**Comment by tvandorffy** [comment]
> Just left it here for reference.

Ah, all right, thanks!

**Comment by berserkov** [comment]
Hi!
Did you have any time to look into support for the N9? It could be a neat and affordable device for tinkering

**Comment by MerlijnWajer** [comment]
I personally haven't, but there has been progress on the N900, we might even have audio calls working soon. So maybe that will incentivise someone to try to work on the N9 port again. I do have a device here, but I don't think this is currently a priority for me personally (over say better userspace)

Does current mainline linux boot on the N9?

**Comment by berserkov** [comment]
Hi Merlijn, 

thanks for the quick reply. Currently I don't have the device, but I will test it with filippz's DT patches, as soon as I can get hold of one. Shouldn't it work much better now with the new PowerVR SGX DDK?

BR

**Comment by MerlijnWajer** [comment]
Yes, I imagine it might work much better now that we can have PowerVR SGX work. Where are these device tree patches that you mention?

**Comment by berserkov** [comment]
Those can be found in the postmarketos kernel config:
https://gitlab.com/postmarketOS/pmaports/-/tree/master/device/testing/linux-nokia-n9

**Comment by berserkov** [comment]
How can I build the rootfs myself? Is there any documentation about it?

**Comment by MerlijnWajer** [comment]
Sorry for the late reply.

arm-sdk and the image-builder are what you'd want to use to build the rootfs and there is some documentation there I think:

https://github.com/maemo-leste/image-builder/
https://github.com/maemo-leste/arm-sdk

There is a script there for the n9; I don't think the image would need to be build there, we'd probably just want to merge in the changes to our droid4-linux (which is also for n900, etc) repository.

**Comment by berserkov** [comment]
Thanks a lot! I will try and report, if I can get the N9 working

**Comment by MerlijnWajer** [comment]
Great, please let us know if you have any questions or run into problems doing so

**Comment by sysdbugfactory** [comment]
hello,

just chiming in to say that I would be happy to be able to install and use maemo leste on my nokia n9.

**Comment by sicelo** [comment]
@berserkov were you able to get started? no rush ... just asking

i also obtained N9 recently. unfortunately N900 has been keeping me busy for a while, but I do plan to look at N9 at some point too
